rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,311 @@
+    exist in rotki's database."""
+    verified_pools = []
+    for pool in all_pools:
+        underlying_token = []
+        # Ensure pool coins exists and underlying tokens in the globaldb.
+        if pool.underlying_token is not None:
+            try:
+                get_or_create_evm_token(
+                    userdb=evm_inquirer.database,
+                    evm_address=pool.underlying_token,
+                    chain_id=evm_inquirer.chain_id,
+                    evm_inquirer=evm_inquirer,
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
+                underlying_token.append(
+                    UnderlyingToken(
+                        address=pool.underlying_token, token_kind=EvmTokenKind.ERC20, weight=ONE,
+                    ),
+                )
+
+            except NotERC20Conformant as e:
+                log.error(
+                    f'Skipping pool {pool} because {pool.underlying_token} is not a '
+                    f'valid ERC20 token. {e}',
+                )
+                continue
+
+        # ensure the lp and farming tokens exist in the db
+        verified_tokens: defaultdict[str, set[str]] = defaultdict(set)
+        for token_type, token_addresses in zip(('pools', 'farm'), (pool.lp_tokens, [pool.farming_pool_token]), strict=False):  # noqa: E501
+            for token_address in token_addresses:
+                try:
+                    token = get_or_create_evm_token(
+                        userdb=evm_inquirer.database,
+                        evm_address=string_to_evm_address(token_address),
+                        chain_id=evm_inquirer.chain_id,
+                        protocol=CPT_GEARBOX,
+                        evm_inquirer=evm_inquirer,
+                        underlying_tokens=underlying_token,
+                        encounter=TokenEncounterInfo(
+                            description='Querying gearbox pools',
+                            should_notify=False,
+                        ),
+                    )
+                    verified_tokens[token_type].add(token.identifier)
+                except NotERC20Conformant as e:
+                    log.error(
+                        f'Skipping pool {pool} because {token_address} is not a '
+                        f'valid ERC20 token. {e}',
+                    )
+                    continue
+
+        pool.farming_pool_token = verified_tokens['farm'].pop()
+        pool.lp_tokens = verified_tokens['pools']
+        verified_pools.append(pool)
+    return verified_pools
+
+
+def get_zapper_tokens(pool_data: list[str], underlying_token_address: ChecksumEvmAddress) -> set[ChecksumEvmAddress]:  # noqa: E501
+    """Extracts zapper tokens from the pool data, excluding the underlying token address
```

Comment:
this function is used only once right? Since is small you can replcae the code directly if that is the case